container/shim/src/fetchers/ipfs-gateway.js

```diff
@@ -0,0 +1,89 @@
+import { cpus } from "node:os";
+import http from "node:http";
+import https from "node:https";
+
+import { streamCAR } from "../utils/car.js";
+import { proxyRequestHeaders, proxyResponseHeaders, toUtf8 } from "../utils/http.js";
+import { debug } from "../utils/logging.js";
+import { IPFS_GATEWAY_ORIGIN } from "../config.js";
+
+const GATEWAY_TIMEOUT = 120_000;
+
+const agentOpts = {
+  keepAlive: true,
+  maxSockets: Math.floor(128 / cpus().length),
```

Comment:
For nodes with over 128 CPUs. Also, lower to get rate limited less

```suggestion
  maxSockets: Math.max(Math.floor(100 / cpus().length), 1),
```